### PR TITLE
dbeaver/dbeaver#17003 add collect logs command

### PR DESCRIPTION
**Pull request by flambchap**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml [hunk 1]
```diff
@@ -94,6 +94,11 @@
     </extension>
 
     <extension point="org.eclipse.ui.commands">
+        <command
+            id="org.jkiss.dbeaver.core.util.collectLogs"
+            categoryId="org.jkiss.dbeaver.core.util"
+            name="Collect Logs"
+            description="Collect log files"/>
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml]: Please add localization

File: plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml [hunk 2]
```diff
@@ -191,6 +196,10 @@
             </activeWhen>
         </handler>
         <handler commandId="org.eclipse.ui.edit.text.openLocalFile" class="org.jkiss.dbeaver.ui.app.standalone.actions.OpenLocalFileActionExt"/>
+        <handler
+            commandId="org.jkiss.dbeaver.core.util.collectLogs"
+            class="org.jkiss.dbeaver.ui.app.standalone.actions.CollectLogsHandler"
+        />
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml:202]:
```suggestion
            class="org.jkiss.dbeaver.ui.app.standalone.actions.CollectLogsHandler"/>
```

**Comment by E1izabeth** [comment]
I propose the next feature for PRO after this PR is merged: to send this zip and the license id to dbeaver support email. What do you think?

File: plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml (hunk 2 above)
Comment (flambchap) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml:202]: Our code style permits both [adding the closing bracket to the last line of a statement](https://github.com/dbeaver/dbeaver/blob/dbeaver/dbeaver%2317003-auto-collect-log-files-from-the-ui/plugins/org.jkiss.dbeaver.cmp.simple.ui/src/org/jkiss/dbeaver/tools/compare/simple/ui/CompareObjectsPageOutput.java#L66-L68) and [putting it on its own line](https://github.com/dbeaver/dbeaver/blob/dbeaver/dbeaver%2317003-auto-collect-log-files-from-the-ui/plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java#L508-L515). I prefer the last option as I find it more readable. Also, it makes it possible to add another line to the bottom of a statement without changing the position of the closing bracket, which sometimes can make a diff clearer.

**Comment by flambchap** [comment]
@E1izabeth Honestly, that would be nice. Unfortunately, there are a lot of caveats to your suggestion that I can't discuss on an open medium (NDA, duh)

File: plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml (hunk 2 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml:202]: But everything around your changes is written in the other style. It's inconsistent now here.

Comment (flambchap) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml:202]: Everything around my changes is written in one line and violates the 140 cap on line length.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/plugin.xml:202]: Oh, yeah, it's not only about the closing bracket. I'm not insisting, just don't like it.

**Pull request closed without merging**